mne/_fiff/reference.py

```diff
@@ -122,12 +121,11 @@ def _check_before_dict_reference(inst, ref_dict):
             )
         # Check that values are not bad channels
         if bad_elem := elem.intersection(bad_ch_set):
-            if elem_name == "value":
-                warn(
-                    f"ref_channels dict contains {elem_name}(s) "
-                    f"({', '.join(bad_elem)}) "
-                    "that are marked as bad channels."
-                )
+            warn(
+                f"ref_channels dict contains {elem_name}(s) "
+                f"({', '.join(bad_elem)}) "
+                "that are marked as bad channels."
+            )
```

Comment:
not sure why this was previously only warning when *values* were marked as bad channels. Seems like trying to re-ref a bad channel is likely indicative of a user mistake and merits a warning. I'll concede it's *less bad* then using a marked-as-bad channel as (one of) the "ref-from" channel(s) though, so I'm open to the idea of doing `logger.info` instead of `warn` in such cases. WDYT?

Reply:
What I had in mind was, say, the original reference (e.g. white matter reference for iEEG) was wrong and some channels become noisy as a consequence of that, and users may want to salvage them by trying out different referencing practices and probably mark them as good afterwards. I agree it's reasonable to have some message to alert the user of possible suboptimal inputs and logger sounds good to me!

Reply:
After second thought such scenarios could be quite rare and a universal warn seems less confusing to future developers